src/Microsoft.DotNet.Compiler.Common/Executable.cs

```diff
@@ -97,13 +97,12 @@ private void MakeCompilationOutputRunnableForCoreCLR()
         {   
             WriteDepsFileAndCopyProjectDependencies(_exporter);
 
-            var emitEntryPoint = _compilerOptions.EmitEntryPoint ?? false;
+            var isRunnable = _compilerOptions.EmitEntryPoint ?? _context.ProjectFile.OverrideIsRunnable;
 
-            if (emitEntryPoint && !_context.IsPortable)
+            if (isRunnable && !_context.IsPortable)
             {
-                // TODO: Pick a host based on the RID
                 VerifyCoreClrPresenceInPackageGraph();
-                CoreHost.CopyTo(_runtimeOutputPath, _compilerOptions.OutputName + Constants.ExeSuffix);
```

Comment:
So publish it does the same as it did before.  I just moved that code into this Executable class.

Here, I changed "build" to do the same thing publish does - get it from the dependencies.

---
In reply to: [71797574](https://github.com/dotnet/cli/pull/3911#discussion_r71797574) [](ancestors = 71797574)